sync_diff_inspector/checkpoints/checkpoints.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checkpoints
+
+import (
+	"container/heap"
+	"context"
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/config"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/report"
+
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/chunk"
+	"github.com/siddontang/go/ioutil2"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+)
+
+const (
+	// SuccessState
+	// for chunk: means this chunk's data is equal
+	// for table: means this all chunk in this table is equal(except ignore chunk)
+	SuccessState = "success"
+
+	// FailedState
+	// for chunk: means this chunk's data is not equal
+	// for table: means some chunks' data is not equal or some chunk check failed in this table
+	FailedState = "failed"
+
+	// for chunk: this chunk is ignored. if sample is not 100% or it is Empty chunk, will ignore some chunk
```

Comment:
There are also some other lint issues, but we can do them in next PR.
```suggestion
	// IgnoreState
	// for chunk: this chunk is ignored. if sample is not 100% or it is Empty chunk, will ignore some chunk
```